website/project/model.py

```diff
@@ -312,6 +340,12 @@ def create(cls, auth, **kwargs):
 
         log_dict.update(comment.root_target.referent.get_extra_log_params(comment))
 
+        if comment.content:
```

Comment:
The check prevents a TypeError if the content is `None`. A validator is used to ensure the content is not none but it doesn't get called until the node is saved.
```
======================================================================
ERROR: test_create_comment_content_cannot_be_none (tests.test_comments.TestCommentModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/laurenbarker/GitHub/COS/osf.io/tests/test_comments.py", line 161, in test_create_comment_content_cannot_be_none
    content=None
  File "/Users/laurenbarker/GitHub/COS/osf.io/website/project/model.py", line 344, in create
    new_mentions = get_valid_mentioned_users_guids(comment, comment.node.contributors)
  File "/Users/laurenbarker/GitHub/COS/osf.io/website/project/model.py", line 200, in get_valid_mentioned_users_guids
    new_mentions = set(re.findall(r"\[[@|\+].*?\]\(\/([a-z\d]{5})\/\)", comment.content))
  File "/Users/laurenbarker/.virtualenvs/osf/lib/python2.7/re.py", line 181, in findall
    return _compile(pattern, flags).findall(string)
TypeError: expected string or buffer
```